doc/nrf/device_guides/nrf91/nrf91_programming.rst

```diff
@@ -0,0 +1,109 @@
+|nRF52840 SoC   |``thingy91/nrf52840``                              |
++---------------+---------------------------------------------------+
+
+You must use the build target ``thingy91/nrf9160/ns`` when building the application code for the nRF9160 SiP and the build target ``thingy91/nrf52840`` when building the application code for the onboard nRF52840 SoC.
+
+.. note::
+
+   * In |NCS| releases before v1.3.0, these build targets were named ``nrf9160_pca20035``, ``nrf9160_pca20035ns``, and ``nrf52840_pca20035``.
+   * In |NCS| releases ranging from v1.3.0 to v1.6.1, the build target ``thingy91/nrf9160/ns`` was named ``thingy91_nrf9160ns``.
+
+.. note::
+
+   LTE/GNSS features can only be used with :ref:`Cortex-M Security Extensions enabled <app_boards_spe_nspe_cpuapp_ns>` (``_ns`` build target).
+
+The table below shows the different types of build files that are generated and the different scenarios in which they are used:
+
++-----------------------+----------------------------------------+----------------------------------------------------------------+
+| File                  | File format                            | Programming scenario                                           |
++=======================+========================================+================================================================+
+|:file:`merged.hex`     | Full image, HEX format                 | Using an external debug probe and nRF Connect Programmer.      |
++-----------------------+----------------------------------------+----------------------------------------------------------------+
+|:file:`app_signed.hex` | MCUboot compatible image, HEX format   | Using the built-in bootloader and nRF Connect Programmer.      |
++-----------------------+----------------------------------------+----------------------------------------------------------------+
+|:file:`app_update.bin` | MCUboot compatible image, binary format|* Using the built-in bootloader and mcumgr command line tool.   |
+|                       |                                        |* For FOTA updates.                                             |
++-----------------------+----------------------------------------+----------------------------------------------------------------+
+
+For an overview of different types of build files in the |NCS|, see :ref:`app_build_output_files`.
+
+There are multiple methods of programming a sample or application onto a Thingy:91.
+It is recommended to use an external debug probe to program the Thingy:91.
+
+.. note::
+
+   If you do not have an external debug probe available to program the Thingy:91, you can directly program by :ref:`using the USB (MCUboot) method and nRF Connect Programmer <programming_thingy>`.
+   In this scenario, use the :file:`app_signed.hex` firmware image file.
+
+.. note::
+
+   While building applications for Thingy:91, the build system changes the signing algorithm of MCUboot so that it uses the default RSA Keys.
```

Comment:
```suggestion
   While building applications for Thingy:91, the build system changes the signing algorithm of MCUboot so that it uses the default RSA keys.
```